### PR TITLE
Interactive Prompts

### PR DESCRIPTION
**Pull request by lukeed**
During `preact create`, if user does not provide the required arguments (`template` and `dest`) we will run them thru the list of ***all*** fields (6 total). This ensures we have all the info we need & exposes the user to all available options that the may have missed.

This is what it looks like:

screen shot 2017-09-05 at 12 14 30 pm

Prompting all fields as a result of the Poll in #352

---

_Closes #358, #352_

### CODE REVIEW

File: src/lib/setup.js [hunk 1]
```diff
@@ -61,3 +61,25 @@ export async function initGit(target) {
...
+export function isMissing(argv) {
+	let out = [];
+
+	const ask = (name, message, val) => {
+		let type = val === void 0 ? 'input' : 'confirm';
+		out.push({ name, message, type, default:val });
+	};
+
+	// Required data
+	!argv.template && ask('template', 'Remote template to clone (user/repo#tag)');
```
Comment (reznord) [inline src/lib/setup.js]: Better to show the default value here. Like what we did for `preact init`?

Comment (reznord) [reply src/lib/setup.js]: basically its like having a fallback value for all the prompts

Comment (lukeed) [reply src/lib/setup.js]: There is no default value -- it's required.

**Comment by lukeed** [comment]
Oh, @reznord, here's another screenshot of the prompt _before_ answering. It shows `N` because `false` is the default here.

screen shot 2017-09-05 at 10 59 22 pm

**Comment by reznord** [comment]
True! That is what we can do for the input fields too right?

**Comment by reznord** [comment]
@lukeed slack'in you (have a few suggestions for this)

**Comment by lukeed** [comment]
> True! That is what we can do for the input fields too right?

For `dest` and `template`, no. We don't want to set _any_ default for those since those are the only two distinct, required fields that the user needs to declare.

For `name`, it's possible, but I don't see a point in doing it. Perhaps `My App` but even then, it's not that useful imo.

**Comment by reznord** [comment]
Makes sense! :)

**Pull request merged**